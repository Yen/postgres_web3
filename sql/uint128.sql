-- TEST -- in function min (success)
SELECT '0'::uint128;
-- TEST -- in function min (success)
SELECT '-0'::uint128;
-- TEST -- in function max (success)
SELECT '340282366920938463463374607431768211455'::uint128;
-- TEST -- in function min - 1 (failure)
SELECT '-1'::uint128;
-- TEST -- in function max + 1 (failure)
SELECT '340282366920938463463374607431768211456'::uint128;

-- min variable
\set uint128_min '\'0\'::uint128'
-- max variable
\set uint128_max '\'340282366920938463463374607431768211455\'::uint128'

-- TEST -- operator add (success)
SELECT 5::uint128 + 2::uint128;
-- TEST -- operator add (failure)
SELECT :uint128_max + 1::uint128;

-- TEST -- operator sub (success)
SELECT 5::uint128 - 2::uint128;
-- TEST -- operator sub (failure)
SELECT :uint128_min - 1::uint128;

-- TEST -- operator mul (success)
SELECT 5::uint128 * 2::uint128;
-- TEST -- operator mul (failure)
SELECT :uint128_max * 2::uint128;

-- TEST -- operator div (success)
SELECT 5::uint128 / 2::uint128;
-- TEST -- operator div (failure)
SELECT 5::uint128 / 0::uint128;

-- TEST -- operator mod (success)
SELECT 5::uint128 % 2::uint128;
-- TEST -- operator mod (failure)
SELECT 5::uint128 % 0::uint128;

-- TEST -- operator lt (success)
SELECT 5::uint128 < 2::uint128;
-- TEST -- operator lt (success)
SELECT 5::uint128 < 5::uint128;

-- TEST -- operator gt (success)
SELECT 5::uint128 > 2::uint128;
-- TEST -- operator gt (success)
SELECT 5::uint128 > 5::uint128;

-- TEST -- operator lteq (success)
SELECT 5::uint128 <= 2::uint128;
-- TEST -- operator lteq (success)
SELECT 5::uint128 <= 5::uint128;

-- TEST -- operator gteq (success)
SELECT 5::uint128 >= 2::uint128;
-- TEST -- operator gteq (success)
SELECT 5::uint128 >= 5::uint128;

-- TEST -- operator eq (success)
SELECT 5::uint128 = 2::uint128;
-- TEST -- operator eq (success)
SELECT 5::uint128 = 5::uint128;

-- TEST -- operator neq (success)
SELECT 5::uint128 <> 2::uint128;
-- TEST -- operator neq (success)
SELECT 5::uint128 <> 5::uint128;

-- TEST -- operator uplus (success)
SELECT +(5::uint128);

-- TEST -- operator uminus (success)
SELECT -(0::uint128);
-- TEST -- operator uminus (failure)
SELECT -(5::uint128);

-- TEST -- operator abs (success)
SELECT @(5::uint128);

-- TEST -- operator bitand (success)
SELECT 5::uint128 & 2::uint128;

-- TEST -- operator bitor (success)
SELECT 5::uint128 | 2::uint128;

-- TEST -- operator bitxor (success)
SELECT 5::uint128 # 2::uint128;

-- TEST -- operator bitnot (success)
SELECT ~(2::uint128);

-- TEST -- operator bitshiftleft (success)
SELECT 5::uint128 << 2;

-- TEST -- operator bitshiftright (success)
SELECT 5::uint128 >> 2;

-- TEST -- cast uint128 as smallint (success)
SELECT 5::uint128::smallint;
-- TEST -- cast uint128 as smallint (failure)
SELECT :uint128_max::smallint;

-- TEST -- cast uint128 as integer (success)
SELECT 5::uint128::integer;
-- TEST -- cast uint128 as integer (failure)
SELECT :uint128_max::integer;

-- TEST -- cast uint128 as bigint (success)
SELECT 5::uint128::bigint;
-- TEST -- cast uint128 as bigint (failure)
SELECT :uint128_max::bigint;

-- TEST -- cast uint128 as int128 (success)
SELECT 5::uint128::int128;
-- TEST -- cast uint128 as int128 (failure)
SELECT :uint128_max::int128;
