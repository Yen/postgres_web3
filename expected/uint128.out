-- TEST -- in function min (success)
SELECT '0'::uint128;
 uint128 
---------
 0
(1 row)

-- TEST -- in function min (success)
SELECT '-0'::uint128;
 uint128 
---------
 0
(1 row)

-- TEST -- in function max (success)
SELECT '340282366920938463463374607431768211455'::uint128;
                 uint128                 
-----------------------------------------
 340282366920938463463374607431768211455
(1 row)

-- TEST -- in function min - 1 (failure)
SELECT '-1'::uint128;
ERROR:  value "-1" is out of range for type uint128
LINE 1: SELECT '-1'::uint128;
               ^
-- TEST -- in function max + 1 (failure)
SELECT '340282366920938463463374607431768211456'::uint128;
ERROR:  value "340282366920938463463374607431768211456" is out of range for type uint128
LINE 1: SELECT '340282366920938463463374607431768211456'::uint128;
               ^
-- min variable
\set uint128_min '\'0\'::uint128'
-- max variable
\set uint128_max '\'340282366920938463463374607431768211455\'::uint128'
-- TEST -- operator add (success)
SELECT '5'::uint128 + '2'::uint128;
 ?column? 
----------
 7
(1 row)

-- TEST -- operator add (failure)
SELECT :uint128_max + '1'::uint128;
ERROR:  uint128 out of range
-- TEST -- operator sub (success)
SELECT '5'::uint128 - '2'::uint128;
 ?column? 
----------
 3
(1 row)

-- TEST -- operator sub (failure)
SELECT :uint128_min - '1'::uint128;
ERROR:  uint128 out of range
-- TEST -- operator mul (success)
SELECT '5'::uint128 * '2'::uint128;
 ?column? 
----------
 10
(1 row)

-- TEST -- operator mul (failure)
SELECT :uint128_max * '2'::uint128;
ERROR:  uint128 out of range
-- TEST -- operator div (success)
SELECT '5'::uint128 / '2'::uint128;
 ?column? 
----------
 2
(1 row)

-- TEST -- operator div (failure)
SELECT '5'::uint128 / '0'::uint128;
ERROR:  division by zero
-- TEST -- operator mod (success)
SELECT '5'::uint128 % '2'::uint128;
 ?column? 
----------
 1
(1 row)

-- TEST -- operator mod (failure)
SELECT '5'::uint128 % '0'::uint128;
ERROR:  division by zero
-- TEST -- operator lt (success)
SELECT '5'::uint128 < '2'::uint128;
 ?column? 
----------
 f
(1 row)

-- TEST -- operator lt (success)
SELECT '5'::uint128 < '5'::uint128;
 ?column? 
----------
 f
(1 row)

-- TEST -- operator gt (success)
SELECT '5'::uint128 > '2'::uint128;
 ?column? 
----------
 t
(1 row)

-- TEST -- operator gt (success)
SELECT '5'::uint128 > '5'::uint128;
 ?column? 
----------
 f
(1 row)

-- TEST -- operator lteq (success)
SELECT '5'::uint128 <= '2'::uint128;
 ?column? 
----------
 f
(1 row)

-- TEST -- operator lteq (success)
SELECT '5'::uint128 <= '5'::uint128;
 ?column? 
----------
 t
(1 row)

-- TEST -- operator gteq (success)
SELECT '5'::uint128 >= '2'::uint128;
 ?column? 
----------
 t
(1 row)

-- TEST -- operator gteq (success)
SELECT '5'::uint128 >= '5'::uint128;
 ?column? 
----------
 t
(1 row)

-- TEST -- operator eq (success)
SELECT '5'::uint128 = '2'::uint128;
 ?column? 
----------
 f
(1 row)

-- TEST -- operator eq (success)
SELECT '5'::uint128 = '5'::uint128;
 ?column? 
----------
 t
(1 row)

-- TEST -- operator neq (success)
SELECT '5'::uint128 <> '2'::uint128;
 ?column? 
----------
 t
(1 row)

-- TEST -- operator neq (success)
SELECT '5'::uint128 <> '5'::uint128;
 ?column? 
----------
 f
(1 row)

-- TEST -- operator uplus (success)
SELECT +('5'::uint128);
 ?column? 
----------
 5
(1 row)

-- TEST -- operator uminus (success)
SELECT -('0'::uint128);
 ?column? 
----------
 0
(1 row)

-- TEST -- operator uminus (failure)
SELECT -('5'::uint128);
ERROR:  uint128 out of range
-- TEST -- operator abs (success)
SELECT @('5'::uint128);
 ?column? 
----------
 5
(1 row)

-- TEST -- operator bitand (success)
SELECT '5'::uint128 & '2'::uint128;
 ?column? 
----------
 0
(1 row)

-- TEST -- operator bitor (success)
SELECT '5'::uint128 | '2'::uint128;
 ?column? 
----------
 7
(1 row)

-- TEST -- operator bitxor (success)
SELECT '5'::uint128 # '2'::uint128;
 ?column? 
----------
 7
(1 row)

-- TEST -- operator bitnot (success)
SELECT ~('2'::uint128);
                ?column?                 
-----------------------------------------
 340282366920938463463374607431768211453
(1 row)

-- TEST -- operator bitshiftleft (success)
SELECT '5'::uint128 << 2;
 ?column? 
----------
 20
(1 row)

-- TEST -- operator bitshiftright (success)
SELECT '5'::uint128 >> 2;
 ?column? 
----------
 1
(1 row)

-- TEST -- cast uint128 as smallint (success)
SELECT '5'::uint128::smallint;
 int2 
------
    5
(1 row)

-- TEST -- cast uint128 as smallint (failure)
SELECT :uint128_max::smallint;
ERROR:  smallint out of range
-- TEST -- cast uint128 as integer (success)
SELECT '5'::uint128::integer;
 int4 
------
    5
(1 row)

-- TEST -- cast uint128 as integer (failure)
SELECT :uint128_max::integer;
ERROR:  integer out of range
-- TEST -- cast uint128 as bigint (success)
SELECT '5'::uint128::bigint;
 int8 
------
    5
(1 row)

-- TEST -- cast uint128 as bigint (failure)
SELECT :uint128_max::bigint;
ERROR:  bigint out of range
-- TEST -- cast uint128 as int128 (success)
SELECT '5'::uint128::int128;
 int128 
--------
 5
(1 row)

-- TEST -- cast uint128 as int128 (failure)
SELECT :uint128_max::int128;
ERROR:  int128 out of range
-- TEST -- cast uint128 as int256 (success)
SELECT '5'::uint128::int256;
 int256 
--------
 5
(1 row)

-- TEST -- cast uint128 as uint256 (success)
SELECT '5'::uint128::uint256;
 uint256 
---------
 5
(1 row)

-- TEST -- cast uint128 as hex160 (success)
SELECT '5'::uint128::hex160;
 hex160 
--------
 5
(1 row)

-- TEST -- cast uint128 as hex256 (success)
SELECT '5'::uint128::hex256;
 hex256 
--------
 5
(1 row)

-- TEST -- cast smallint as uint128 (success)
SELECT '5'::smallint::uint128;
 uint128 
---------
 5
(1 row)

-- TEST -- cast smallint as uint128 (failure)
SELECT '-5'::smallint::uint128;
ERROR:  uint128 out of range
-- TEST -- cast integer as uint128 (success)
SELECT '5'::integer::uint128;
 uint128 
---------
 5
(1 row)

-- TEST -- cast integer as uint128 (failure)
SELECT '-5'::integer::uint128;
ERROR:  uint128 out of range
-- TEST -- cast bigint as uint128 (success)
SELECT '5'::bigint::uint128;
 uint128 
---------
 5
(1 row)

-- TEST -- cast bigint as uint128 (failure)
SELECT '-5'::bigint::uint128;
ERROR:  uint128 out of range
